db/models/user/user.go

```diff
@@ -0,0 +1,117 @@
+package user
+
+import (
+	"database/sql"
+	"errors"
+	"strconv"
+	"time"
+
+	"tweeter/db"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+// User represents a user of the application
+type User struct {
+	ID       ID `db:"user_id"`
+	Email    string
+	Password string
+	Created  time.Time
+	Modified time.Time
+}
+
+// MinimumPasswordLength is the minimum length of a password
+var MinimumPasswordLength = 6
+
+// ErrUserNotFound is the error returned when the given user does not exist
+var ErrUserNotFound = errors.New("could not find user from information given")
+
+// ErrInternalError is the error returned when normal flow failed (db error, timeout, etc)
+var ErrInternalError = errors.New("internal error, retry later")
+
+// ErrMismatchedPassword is the error returned by Validate when password do not match stored password
+var ErrMismatchedPassword = errors.New("invalid password given")
+
+// ErrPasswordTooShort is the error returned by Create when the password is too short
+var ErrPasswordTooShort = errors.New("password too short")
+
+// ErrUserEmailAlreadyExists is the error returned by Create when the email already exists
+var ErrUserEmailAlreadyExists = errors.New("user already exists for email")
+
+// ID is a type alias for the user's ID type
+type ID uint64
+
+// ParseID parses a string into a ID
+func ParseID(idString string) (ID, error) {
```

Comment:
If this error is returned to the user externally (which it is), we should use a pre-defined error like the others above.